src/sdk/account/account.service.ts

```diff
@@ -232,15 +232,15 @@ export class AccountService extends Service {
     return result;
   }
 
-  async getAccountBalances(account: string, tokens: string[], ChainId: number): Promise<AccountBalances> {
+  async getAccountBalances(account: string, tokens: string[], ChainId: number, provider?: string): Promise<AccountBalances> {
     const { apiService } = this.services;
 
     const { result } = await apiService.query<{
       result: AccountBalances;
     }>(
       gql`
-        query($ChainId: Int!, $account: String!, $tokens: [String!]) {
-          result: accountBalances(chainId: $ChainId, account: $account, tokens: $tokens) {
+        query($ChainId: Int!, $account: String!, $tokens: [String!], $provider: String) {
```

Comment:
Isn't there a need to add a check to see if Provider has a valid value before passing to GQL? What happens in the case of an empty string?

Reply:
It will consider null and follow the default route. The same goes for an invalid strings such as **zapper1 or Moralis123**